src/Plutarch/Test/Export.hs

```diff
@@ -0,0 +1,728 @@
+  case lookedUp of
+    Nothing -> tell . noSuchScript $ name
+    Just envelope -> case view (to tsRole) envelope of
+      ValidatorRole -> do
+        datum' <- asks (toData . view #datum)
+        redeemer' <- asks (toData . view #redeemer)
+        sc' <- asks (toData . view #context)
+        let script = view (to tsScript) envelope
+        let result = AppliedValidator script datum' redeemer' sc'
+        tell . prepareTest name reason outcome $ result
+      MintingPolicyRole -> tell . roleMismatch name reason $ ValidatorRole
+
+{- | As 'assertValidator', but for a minting policy instead.
+
+ @since 1.2.1
+-}
+assertMintingPolicy ::
+  forall (dat :: Type) (red :: Type) (info :: Type).
+  (ToData dat) =>
+  -- | What you expect to happen
+  ScriptOutcome ->
+  -- | The reason or explanation to label the test with
+  String ->
+  -- | The name of the script to run
+  Text ->
+  WithExport dat red info ()
+assertMintingPolicy outcome reason name = WithExport $ do
+  lookedUp <- asks (view (#raws % #rawScripts % at' @(Map _ _) name))
+  case lookedUp of
+    Nothing -> tell . noSuchScript $ name
+    Just envelope -> case tsRole envelope of
+      ValidatorRole -> tell . roleMismatch name reason $ MintingPolicyRole
+      MintingPolicyRole -> do
+        datum' <- asks (toData . view #datum)
+        sc' <- asks (toData . view #context)
+        let script = tsScript envelope
+        let result = AppliedMP script datum' sc'
+        tell . prepareTest name reason outcome $ result
+
+{- | Modify the given test collection to use a different datum, based on the
+ provided function. The additional information provided by the script export
+ is provided as an additional parameter to the function argument.
+
+ = Example
+
+ > tests = do
+ >    -- This will use unmodified arguments
+ >    assertValidator Crashes "bad datum" "myScript"
+ >    modifyingDatum fixMyDatum $ do
+ >      -- This will use a datum modified according to fixMyDatum
+ >      assertValidator Runs "good datum" "myScript"
+ >      -- So will this
+ >      assertValidator Crashes "this one is wrong" "myScript2"
+ >    -- However, here, the change resets
+ >    assertValidator Runs "but this one is good" "myScript2"
+
+ You can \'nest\' modifications, which behave according to lexical scope:
+
+ > tests = do
+ >    -- One change
+ >    modifyingDatum change1 $ do
+ >      -- This will only modify with change1
+ >      assertValidator Crashes "not enough fixes" "myScript"
+ >      -- Stack on another change
+ >      modifyingDatum change2 $ do
+ >        -- This modifies with change1, then change2 on top
+ >        assertValidator Runs "enough fixes" "myScript"
+ >      -- reset change2
+ >      assertValidator Crashes "why did we undo this change?" "myScript"
+ >   -- reset change1
+
+ @since 1.2.1
+-}
+modifyingDatum ::
+  forall (dat :: Type) (red :: Type) (info :: Type) (a :: Type).
+  -- | How to \'locally modify\' the datum
+  (info -> dat -> dat) ->
```

Comment:
How useful would passing `info` to modifier? I think `modifyingDatum'` should be the default.

Reply:
same goes for other modifiers and setters

Reply:
In oracles at least this is _very_ useful for script context modifications.